cinder/volume/drivers/lightos.py

```diff
@@ -1091,12 +1092,41 @@ def __add_volume_acl(self, project_name, lightos_volname, acl_to_add, host_ips):
         if acl_to_add not in acl:
            acl.append(acl_to_add)
 
-        ip_acl=[]    
+        if 'ALLOW_NONE' in ip_acl:
+            ip_acl.remove('ALLOW_NONE')
+        
         if self.use_ip_acl():
-           ip_acl=host_ips
+            ip_acl = list(set(ip_acl).union(set(host_ips)))
         else:
            ip_acl.append('ALLOW_ANY')
-                
+
+        # The max (16) elemenets are allowed in IPACL.
+        # if elements are more than 16 then remove 
+        # less-frequently used IPv6 address(s), and IPv4 if needed.     
+               
+        ipv4addrs = [addr for addr in ip_acl if netutils.is_valid_ipv4(addr)]
+        ipv6addrs = [addr for addr in ip_acl if netutils.is_valid_ipv6(addr)]
+        
+        IpAcl_size = 16
+        if len(ipv4addrs) > IpAcl_size:
```

Comment:
@rahman-lb what is the implication of "silently"  droping the ipv4headers?
for specific use case we have in mind, limit is likely not relavent, so no implication.
but assuming we have more than 16, than one of the hosts will now loose connectivity?
shouldnt we be returning a clearer error, failing the operation?

p.s is this function invoked from both create and update (attach) volume?

Reply:
@ronen-lb  This is very unlikely for one physical server to have more than 16 Physical nics and configured with IPv4.  Most of the server will have at max 6-8 Physical nics. I believe dropping IPv4 will be rare in real environment, it's a just safety check better than failing the entire process. 
Attache will not have this problem because during the attachment process volume IP-ACL will be empty and IPs are written from only one server.

Reply:
@rahman-lb i agree that this is not highly likely, however you intentionally added code to handle this case.
i am concerned that dropping the IPv4 silently would be much harder to understand and debug the consequences of rather than failing the operation

Reply:
@ronen-lb I understand your concern, Muli expressed his concern too but  in this case the chances of failure will be 1/16 otherwise it be a guaranteed failure. 
If you think we  should break and throw error instead of ignoring IPv4, sure we can make that change when we're creating upstream PR.